### DISCUSSION TITLE
spring-cloud-aws-messaging in 3.0

### DISCUSSION THREAD

**Discussion by iparadiso** [awspring/spring-cloud-aws#628] [General]
Hi folks.  I'm testing out some spring boot 3.0 migration and `spring-coud-aws-messaging` from 2.4.2 is no more in 3.0.0-M3.  What's the recommended migration for using `QueueMessagingTemplate` and related classes?  I do not see a straight forward migration in `spring-cloud-aws-sqs` or `spring-cloud-aws-core`.  Did these classes get moved somewhere else?  I couldn't find anything in the release notes.

**Comment by tomazfernandes** [comment]
Hi @iparadiso, we're currently [working on that](https://github.com/awspring/spring-cloud-aws/pull/597).

It should be ready and released in the `3.0.0-RC` in a few days.

In the meantime if you usage is simple enough you can create a simple class such as [this](https://docs.awspring.io/spring-cloud-aws/docs/3.0.0-SNAPSHOT/reference/html/index.html#sending-messages).

Hope that helps!

**Reply by iparadiso** [reply]
Thanks @tomazfernandes.  Looking forward to the RC.

**Reply by mbode-asys** [reply]
Hi @tomazfernandes any news related to 3.0.0-RC2? I can only find 2.4.4.

**Reply by tomazfernandes** [reply]
Hi @mbode-asys, RC2 has been released a few weeks back, and is available on Maven Central.

**Reply by mbode-asys** [reply]
> Hi @mbode-asys, RC2 has been released a few weeks back, and is available on Maven Central.

Hi @tomazfernandes thanks for your answers. And you are right. 
Actually I puzzled the artifacts. I search for spring-cloud-starter-aws-messaging 3.0.0-RC2 and the latest I can locate is 2.4.4.

Has this moved into one of the other 3.0.0-RC2 artifacts? All of the following contain a 3.0.0-RC2:
- [spring-cloud-aws-starter/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter/)                                          
- [spring-cloud-aws-starter-dynamodb/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-dynamodb/)                               
- [spring-cloud-aws-starter-metrics/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-metrics/)                                
- [spring-cloud-aws-starter-parameter-store/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-parameter-store/)
- [spring-cloud-aws-starter-s3/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-s3/)                            
- [spring-cloud-aws-starter-secrets-manager/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-secrets-manager/)
- [spring-cloud-aws-starter-ses/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-ses/)                                    
- [spring-cloud-aws-starter-sns/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-sns/)                                    
- [spring-cloud-aws-starter-sqs/](https://repo1.maven.org/maven2/io/awspring/cloud/spring-cloud-aws-starter-sqs/)                                    

Many thanks.

**Comment by CasaSky** [comment]
Hi @tomazfernandes i just saw that the RC is been reverted to snapshot. Can a RC be used in production or should we wait for the final release?

**Reply by maciejwalkowiak** [reply]
It is not reverted to snapshot but a common practice, once RC is released, version is set back to snapshots. The only difference between RC and GA is that we may slightly change some APIs, so keep that in mind, but other than that I would go for it.

**Reply by CasaSky** [reply]
Thanks for the quick response and the explanation. Everything makes sense now. We'll asap migrate to 3.0.0-RC1. Good luck for the GA.

**Comment by iparadiso** [comment]
HI @tomazfernandes I'm looking at the level of effort to migrate to 3.0 RC1, and the biggest impact I've seen so far is `QueueMessagingTemplate` --> `SqsTemplate`.  We have hundreds of usages across hundreds of repos.  Has your team considered contributing a recipe for some of these heavier lifts to OpenRewrite?  They have a [family of recipes for spring boot 3.x](https://docs.openrewrite.org/reference/recipes/java/spring/boot3) that looks promising with some additional contributions.

**Reply by maciejwalkowiak** [reply]
@iparadiso we haven't but if you would like to help us with that we are open
